src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -908,96 +911,148 @@ private List<Connection> getCongestedConnections() {
+                assert(existing == null);
+            }
+            LUTTools.swapMultipleLutPins(pinSwaps);
+        }
+
+        if (lutRoutethru) {
+            // When LUT routethru-s are considered, examine both static nets to find
+            // any cases where the *MUX output pin is used as a static source alongside
+            // the *_O output being used as a routethru. In such cases, configure the
+            // OUTMUX* site PIP to source from the 5LUT rather than the default 6LUT
+            // so that no conflict occurs
+            for (Net staticNet : Arrays.asList(design.getGndNet(), design.getVccNet())) {
+                for (SitePinInst spi : staticNet.getPins()) {
+                    if (!spi.isOutPin()) {
+                        continue;
+                    }
+                    SiteInst si = spi.getSiteInst();
+                    if (!Utils.isSLICE(si)) {
+                        continue;
+                    }
+
+                    String pinName = spi.getName();
+                    if (!pinName.endsWith("MUX")) {
                         continue;
                     }
 
-                    List<RouteNode> rnodes = connection.getRnodes();
-                    RouteNode newSinkRnode = rnodes.get(0);
-                    if (newSinkRnode == connection.getSinkRnode()) {
+                    Node muxNode = spi.getConnectedNode();
+                    assert(routingGraph.getPreservedNet(muxNode) == staticNet);
+
+                    Site site = si.getSite();
+                    char lutLetter = pinName.charAt(0);
+                    Node oNode = site.getConnectedNode(lutLetter + "_O");
+                    RouteNode rnode = routingGraph.getNode(oNode);
+                    if (rnode == null || rnode.getOccupancy() == 0) {
+                        // No LUT6 routethru, nothing to be done
                         continue;
                     }
-                    connection.setSinkRnode(newSinkRnode);
 
-                    SitePin newSitePin = newSinkRnode.getNode().getSitePin();
-                    String existing = pinSwaps.put(oldSinkSpi, newSitePin.getPinName());
-                    assert(existing == null);
+                    // Only expect GND net to use SLICE outputs
+                    assert(staticNet.getType() == NetType.GND);
+
+                    // Perform intra-site routing back to the LUT5 to not conflict with LUT6 routethru
+                    BEL outmux = si.getBEL("OUTMUX" + lutLetter);
+                    si.routeIntraSiteNet(staticNet, outmux.getPin("D5"), outmux.getPin("OUT"));
+
+                    if (si.getDesign() == null) {
+                        // Rename SiteInst (away from "STATIC_SOURCE_<siteName>") and
+                        // attach it to the design so that intra-site routing updates take effect
+                        si.setName(site.getName());
+                        design.addSiteInst(si);
```

Comment:
Can you demonstrate that this doesn't cause an error when importing the designs back into Vivado?  Unless a placed cell exists (which a route thru may suffice), my understanding is that this could lead to an issue.

Reply:
I'm seeing a breakpoint hit this line in the newly added test`testNonTimingDrivenFullRoutingWithLutRoutethru` with `bnn.dcp` for `SLICE_X79Y201`, and for `optical-flow.dcp` for `SLICE_X66Y252` and 10 others.

For `bnn.dcp`, even after RWRoute is done, that `SiteInst` has no placed cells, yet Vivado 2022.1 seems to accept it just fine and reports a clean routing result.

![image](https://github.com/Xilinx/RapidWright/assets/90657806/17f5b32b-1e60-441a-8cf2-9605ba15cc05)
(highlighted net is `GLOBAL_LOGIC0`; rest of slice is completely empty)

Vivado 2022.1 also accepts this `optical-flow.dcp` routed result.

Reply:
We probably need to upgrade our terminology as there are "routethrus" that terminate at a cell in the site and 
other "routethrus" that don't (being specified by a `PIP`).